File: sportx_app/lib/features/talent_scout/data/models/talent_scout_profile.dart
class TalentScoutProfile {
  final int? id;
  final int userId;
  final String? organization;
  final String? affiliation;
  final List<int> sportsSpecialization;
  final int? experienceYears;
  final int? cityId;
  final String? bio;
  final int? photoMediaId;
  final String? photoUrl;
  final bool listingStatus;

  TalentScoutProfile({
    this.id,
    required this.userId,
    this.organization,
    this.affiliation,
    this.sportsSpecialization = const [],
    this.experienceYears,
    this.cityId,
    this.bio,
    this.photoMediaId,
    this.photoUrl,
    this.listingStatus = true,
  });

  factory TalentScoutProfile.fromJson(Map<String, dynamic> json) {
    final photo = json['photo'] as Map<String, dynamic>?;
    return TalentScoutProfile(
      id: json['id'] as int?,
      userId: json['user_id'] as int,
      organization: json['organization'] as String?,
      affiliation: json['affiliation'] as String?,
      sportsSpecialization: (json['sports_specialization'] as List<dynamic>?)
              ?.map((e) => (e as num).toInt())
              .toList() ??
          [],
      experienceYears: (json['experience_years'] as num?)?.toInt(),
      cityId: (json['city_id'] as num?)?.toInt(),
      bio: json['bio'] as String?,
      photoMediaId: (json['photo_media_id'] as num?)?.toInt(),
      photoUrl: photo?['url'] as String?,
      listingStatus: json['listing_status'] == true || json['listing_status'] == 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'organization': organization,
      'affiliation': affiliation,
      'sports_specialization': sportsSpecialization,
      'experience_years': experienceYears,
      'city_id': cityId,
      'bio': bio,
      if (photoMediaId != null) 'photo_media_id': photoMediaId,
      'listing_status': listingStatus,
    };
  }
}

File: sportx-api/app/Models/TalentScoutProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Traits\Auditable;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class TalentScoutProfile extends Model
{
    use Auditable;
    use SoftDeletes;

    protected $fillable = [
        'user_id', 'organization', 'affiliation',
        'sports_specialization', 'experience_years',
        'city_id', 'bio', 'photo_media_id', 'listing_status',
    ];

    protected $casts = [
        'sports_specialization' => 'array',
        'listing_status' => 'boolean',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function city(): BelongsTo
    {
        return $this->belongsTo(City::class);
    }

    public function photo(): BelongsTo
    {
        return $this->belongsTo(MediaItem::class, 'photo_media_id');
    }

    public function shortlists(): HasMany
    {
        return $this->hasMany(ScoutShortlist::class);
    }

    public function connections(): HasMany
    {
        return $this->hasMany(ScoutConnection::class);
    }
}
